src/draw/lv_img_decoder.c

```diff
@@ -274,445 +169,15 @@ void lv_img_decoder_set_close_cb(lv_img_dec_t * decoder, lv_img_decoder_close_f_
     decoder->close_cb = close_cb;
 }
 
-/**
- * Get info about a built-in image
- * @param decoder the decoder where this function belongs
- * @param src the image source: pointer to an `lv_img_dsc_t` variable, a file path or a symbol
- * @param header store the image data here
- * @return LV_RES_OK: the info is successfully stored in `header`; LV_RES_INV: unknown format or other error.
- */
-lv_res_t lv_img_decoder_built_in_info(lv_img_dec_t * decoder, const void * src, lv_img_header_t * header)
+bool lv_img_decoder_has_size_hint(const lv_img_dec_dsc_in_t * dsc)
 {
-    LV_UNUSED(decoder); /*Unused*/
-
-    lv_img_src_type_t src_type = lv_img_src_get_type(src);
-
-    if(src_type == LV_IMG_SRC_VARIABLE) {
-        lv_img_cf_t cf = ((lv_img_dsc_t *)src)->header.cf;
-        if(cf < CF_BUILT_IN_FIRST || cf > CF_BUILT_IN_LAST) return LV_RES_INV;
-
-        header->w  = ((lv_img_dsc_t *)src)->header.w;
-        header->h  = ((lv_img_dsc_t *)src)->header.h;
-        header->cf = ((lv_img_dsc_t *)src)->header.cf;
-    }
-    else if(src_type == LV_IMG_SRC_FILE) {
-        /*Support only "*.bin" files*/
-        if(strcmp(lv_fs_get_ext(src), "bin")) return LV_RES_INV;
-
-        lv_fs_file_t f;
-        lv_fs_res_t res = lv_fs_open(&f, src, LV_FS_MODE_RD);
-        if(res == LV_FS_RES_OK) {
-            uint32_t rn;
-            res = lv_fs_read(&f, header, sizeof(lv_img_header_t), &rn);
-            lv_fs_close(&f);
-            if(res != LV_FS_RES_OK || rn != sizeof(lv_img_header_t)) {
-                LV_LOG_WARN("Image get info get read file header");
-                return LV_RES_INV;
-            }
-        }
-
-        if(header->cf < CF_BUILT_IN_FIRST || header->cf > CF_BUILT_IN_LAST) return LV_RES_INV;
-    }
-    else if(src_type == LV_IMG_SRC_SYMBOL) {
-        /*The size depend on the font but it is unknown here. It should be handled outside of the
-         *function*/
-        header->w = 1;
-        header->h = 1;
-        /*Symbols always have transparent parts. Important because of cover check in the draw
-         *function. The actual value doesn't matter because lv_draw_label will draw it*/
-        header->cf = LV_IMG_CF_ALPHA_1BIT;
-    }
-    else {
-        LV_LOG_WARN("Image get info found unknown src type");
-        return LV_RES_INV;
-    }
-    return LV_RES_OK;
+    /* Invalid is if both set to LV_SIZE_CONTENT or if any is 0*/
+    return !((dsc->size_hint.x == LV_SIZE_CONTENT && dsc->size_hint.y == LV_SIZE_CONTENT)
```

Comment:
Just for better readability. 
```suggestion
    if(dsc->size_hint.x == 0) return false;
    if(dsc->size_hint.y == 0) return false; 
    if(dsc->size_hint.x == LV_SIZE_CONTENT && dsc->size_hint.y == LV_SIZE_CONTENT) return false;
    return true;
```